api/searchCases/index.ts

```diff
@@ -144,14 +178,20 @@ export function handleElasticSearchResponse(proxyRes, req, res, json): {} {
       return caseObj
     })
 
-    const handledResponse = {
+    return {
       'columns': json.headers[0].fields,
       'results': results,
       'total': json.total,
     }
-
-    return handledResponse
-  } else {
-    return {}
   }
+  return {}
+}
+
+function canApplyWildCardSearch(wildcardSearchFields: { [key: string]: string[] }, caseType: string, criterion: string): boolean {
+  return wildcardSearchFields.hasOwnProperty(caseType) && wildcardSearchFields[caseType].indexOf(criterion) >= 0
```

Comment:
This doesn't handle a scenario where `wildcardSearchFields[caseType] === undefined` (or indeed `=== null`.

Reply:
Very defensive code but will add it.

Reply:
Many a nasty bug has been avoided by appropriately defensive code. 😁

Reply:
Actually, it also doesn't handle `wildcardSearchFields` itself being `null` or `undefined`...